code/game/objects/items/explosives/mine.dm

```diff
@@ -136,80 +212,204 @@ Stepping directly on the mine will also blow it up
 	var/obj/item/card/id/id = L.get_idcard()
 	if(id?.iff_signal & iff_signal)
 		return FALSE
+	trigger_explosion()
 
-	L.visible_message(span_danger("[icon2html(src, viewers(L))] \The [src] clicks as [L] moves in front of it."), \
-	span_danger("[icon2html(src, viewers(L))] \The [src] clicks as you move in front of it."), \
-	span_danger("You hear a click."))
-
-	playsound(loc, 'sound/weapons/mine_tripped.ogg', 25, 1)
-	INVOKE_ASYNC(src, PROC_REF(trigger_explosion))
-	return TRUE
-
-/// Alien attacks trigger the explosive to instantly detonate
-/obj/item/explosive/mine/attack_alien(mob/living/carbon/xenomorph/X, damage_amount = X.xeno_caste.melee_damage, damage_type = BRUTE, damage_flag = "", effects = TRUE, armor_penetration = 0, isrightclick = FALSE)
-	if(X.status_flags & INCORPOREAL)
-		return FALSE
-	if(triggered) //Mine is already set to go off
-		return
-
-	if(X.a_intent == INTENT_HELP)
-		return
-	X.visible_message(span_danger("[X] has slashed [src]!"), \
-	span_danger("We slash [src]!"))
-	playsound(loc, 'sound/weapons/slice.ogg', 25, 1)
-	INVOKE_ASYNC(src, PROC_REF(trigger_explosion))
-
-/// Trigger an actual explosion and delete the mine.
+///Trigger the mine; needs to be a separate proc so that we can use a timer
 /obj/item/explosive/mine/proc/trigger_explosion()
 	if(triggered)
-		return
+		return FALSE
 	triggered = TRUE
-	explosion(tripwire ? tripwire.loc : loc, light_impact_range = 3)
-	QDEL_NULL(tripwire)
+	visible_message(span_danger("[icon2html(src, viewers(src))] \The [src] [detonation_message]."))
+	playsound(loc, detonation_sound, 50, sound_range = 7)
+	if(detonation_delay)
+		return addtimer(CALLBACK(src, PROC_REF(explode)), detonation_delay)
+	explode()
+
+///Proc that actually causes the explosion
+/obj/item/explosive/mine/proc/explode()
+	if(!triggered)
```

Comment:
what if mine can trigger multiple times?

Reply:
That is what the reusable var is for.